js/ad_manager_vast.js

```diff
@@ -618,16 +769,20 @@ OO.Ads.manager(function(_, $) {
      * @method Vast#adVideoPlaying
      */
     this.adVideoPlaying = function() {
-      if (currentAd && currentAd.ad && currentAd.ad.nextAdInPod) {
+      if (currentAd && currentAd.ad && currentAd.ad.nextAdInPod && !fromPause) {
         var metadata = currentAd.ad.nextAdInPod;
         if (metadata) {
           var ad = generateAd(metadata);
 
-          if (metadata.streamUrl != null || (ad.adType == this.amc.ADTYPE.LINEAR_VIDEO && !_.isEmpty(metadata.streams))) {
+          if (metadata.streamUrl != null ||
+              (ad.adType == this.amc.ADTYPE.LINEAR_VIDEO && !_.isEmpty(metadata.streams)) ||
+              currentAd.ad.data.adType === 'vpaid') {
```

Comment:
Can we introduce a utility function to check if an ad is a vpaid or not? I see this check used a few times.